include/flatbuffers/flatbuffers.h

```diff
@@ -685,6 +709,16 @@ FLATBUFFERS_FINAL_CLASS
     if (string_pool) delete string_pool;
   }
 
+  void Reset() {
```

Comment:
We could reset the vector after we clear, which would have the same end result. Un-optimized, there'd be an extra store to the `cur_` pointer, but I _think_ the compiler should be able to eliminate that first store (correct me if I'm wrong). I'll update to do so, since either way the cost is minimal/zero.